bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/Utils.java

```diff
@@ -359,17 +362,17 @@ public static Resource getWrappedImageResourceWithInheritance(Resource resource,
 
             if (StringUtils.isNotEmpty(linkURL)) {
                 // the inherited resource is the featured image of the linked page
-                Optional<Link> link = linkHandler.getLink(resource);
+                Optional<Link> link = Optional.of(linkHandler.get(resource).build());
```

Comment:
be careful with code like this (only commenting this here, other places affected as well): linkHandler.get never returns null, so wrapping it in Optional is quite useless. instead it leads to false assumption that and invalid link produces any empty optional (which is not the case). instead, it returns a link object that is marked with valid=false. this needs to be checked for further link processing, instead of Optional. or create a helper method for the legacy code that converts a link object o an Optional, setting it to empty if the link is not valid.

Reply:
I introduced a helper method `getOptionalLink()` at https://github.com/adobe/aem-core-wcm-components/pull/2224/commits/624c108ed14f56dd2b361b645f68368129866033